processor/processor.go

```diff
@@ -80,18 +80,13 @@ type trackedUsersReporter interface {
 	GenerateReportsFromJobs(jobs []*jobsdb.JobT, sourceIdFilter map[string]bool) []*trackedusers.UsersReport
 }
 
-type warehouseTransformer interface {
-	transformer.DestinationTransformer
-	CompareAndLog(events []types.TransformerEvent, pResponse, wResponse types.Response, metadata *types.Metadata, eventsByMessageID map[string]types.SingularEventWithReceivedAt)
```

Comment:
Shouldn't we keep compareAndLog until the feature has been rolled out to all customers and we are confident about its correctness?

Reply:
This PR, I will merge this in the end once the correctness has been guaranteed and we can start using the new codebase.